BUG: tests fail with jaxtyping==0.2.24

# Bug description

Tests fail with the latest version of `jaxtyping`:

## Environment
I am running tests from a freshly created venv, with the `equinox` git checked out at `v0.11.2`.

```
❯ ./venv/bin/pip list            
Package           Version
----------------- -------
beartype          0.16.4
equinox           0.11.2
iniconfig         2.0.0
jax               0.4.20
jaxlib            0.4.20
jaxtyping         0.2.24
ml-dtypes         0.3.1
numpy             1.26.2
opt-einsum        3.3.0
packaging         23.2
pip               23.3.1
pluggy            1.3.0
pytest            7.4.3
scipy             1.11.4
setuptools        65.5.0
typeguard         2.13.3
typing_extensions 4.8.0
```

## Error log
```
============================================================================== short test summary info ===============================================================================
ERROR tests/test_abstract.py - beartype.roar.BeartypeDecorHintNonpepException: Method equinox.nn._batch_norm.check_return() parameter "key" type hint <function PRNGKey at 0x7f330062aac0> either PEP-noncomplia...
ERROR tests/test_ad.py - beartype.roar.BeartypeDecorHintNonpepException: Method equinox.nn._batch_norm.check_return() parameter "key" type hint <function PRNGKey at 0x7f330062aac0> either PEP-noncomplia...
ERROR tests/test_caches.py - beartype.roar.BeartypeDecorHintNonpepException: Method equinox.nn._batch_norm.check_return() parameter "key" type hint <function PRNGKey at 0x7f330062aac0> either PEP-noncomplia...
ERROR tests/test_callback.py - beartype.roar.BeartypeDecorHintNonpepException: Method equinox.nn._batch_norm.check_return() parameter "key" type hint <function PRNGKey at 0x7f330062aac0> either PEP-noncomplia...
ERROR tests/test_checkpoint.py - beartype.roar.BeartypeDecorHintNonpepException: Method equinox.nn._batch_norm.check_return() parameter "key" type hint <function PRNGKey at 0x7f330062aac0> either PEP-noncomplia...
ERROR tests/test_closure_to_pytree.py
ERROR tests/test_debug.py - beartype.roar.BeartypeDecorHintNonpepException: Method equinox.nn._batch_norm.check_return() parameter "key" type hint <function PRNGKey at 0x7f330062aac0> either PEP-noncomplia...
ERROR tests/test_enum.py - beartype.roar.BeartypeDecorHintNonpepException: Method equinox.nn._batch_norm.check_return() parameter "key" type hint <function PRNGKey at 0x7f330062aac0> either PEP-noncomplia...
ERROR tests/test_errors.py - beartype.roar.BeartypeDecorHintNonpepException: Method equinox.nn._batch_norm.check_return() parameter "key" type hint <function PRNGKey at 0x7f330062aac0> either PEP-noncomplia...
ERROR tests/test_eval_shape.py - beartype.roar.BeartypeDecorHintNonpepException: Method equinox.nn._batch_norm.check_return() parameter "key" type hint <function PRNGKey at 0x7f330062aac0> either PEP-noncomplia...
ERROR tests/test_filters.py - beartype.roar.BeartypeDecorHintNonpepException: Method equinox.nn._batch_norm.check_return() parameter "key" type hint <function PRNGKey at 0x7f330062aac0> either PEP-noncomplia...
ERROR tests/test_finalise_jaxpr.py - beartype.roar.BeartypeDecorHintNonpepException: Method equinox.nn._batch_norm.check_return() parameter "key" type hint <function PRNGKey at 0x7f330062aac0> either PEP-noncomplia...
ERROR tests/test_jit.py - beartype.roar.BeartypeDecorHintNonpepException: Method equinox.nn._batch_norm.check_return() parameter "key" type hint <function PRNGKey at 0x7f330062aac0> either PEP-noncomplia...
ERROR tests/test_make_jaxpr.py - beartype.roar.BeartypeDecorHintNonpepException: Method equinox.nn._batch_norm.check_return() parameter "key" type hint <function PRNGKey at 0x7f330062aac0> either PEP-noncomplia...
ERROR tests/test_misc.py - beartype.roar.BeartypeDecorHintNonpepException: Method equinox.nn._batch_norm.check_return() parameter "key" type hint <function PRNGKey at 0x7f330062aac0> either PEP-noncomplia...
ERROR tests/test_module.py - beartype.roar.BeartypeDecorHintNonpepException: Method equinox.nn._batch_norm.check_return() parameter "key" type hint <function PRNGKey at 0x7f330062aac0> either PEP-noncomplia...
ERROR tests/test_nn.py - beartype.roar.BeartypeDecorHintNonpepException: Method equinox.nn._batch_norm.check_return() parameter "key" type hint <function PRNGKey at 0x7f330062aac0> either PEP-noncomplia...
ERROR tests/test_noinline.py - beartype.roar.BeartypeDecorHintNonpepException: Method equinox.nn._batch_norm.check_return() parameter "key" type hint <function PRNGKey at 0x7f330062aac0> either PEP-noncomplia...
ERROR tests/test_nontraceable.py - beartype.roar.BeartypeDecorHintNonpepException: Method equinox.nn._batch_norm.check_return() parameter "key" type hint <function PRNGKey at 0x7f330062aac0> either PEP-noncomplia...
ERROR tests/test_onnx.py - beartype.roar.BeartypeDecorHintNonpepException: Method equinox.nn._batch_norm.check_return() parameter "key" type hint <function PRNGKey at 0x7f330062aac0> either PEP-noncomplia...
ERROR tests/test_pformat.py - beartype.roar.BeartypeDecorHintNonpepException: Method equinox.nn._batch_norm.check_return() parameter "key" type hint <function PRNGKey at 0x7f330062aac0> either PEP-noncomplia...
ERROR tests/test_pmap.py - beartype.roar.BeartypeDecorHintNonpepException: Method equinox.nn._batch_norm.check_return() parameter "key" type hint <function PRNGKey at 0x7f330062aac0> either PEP-noncomplia...
ERROR tests/test_primitive.py - beartype.roar.BeartypeDecorHintNonpepException: Method equinox.nn._batch_norm.check_return() parameter "key" type hint <function PRNGKey at 0x7f330062aac0> either PEP-noncomplia...
ERROR tests/test_scan.py - beartype.roar.BeartypeDecorHintNonpepException: Method equinox.nn._batch_norm.check_return() parameter "key" type hint <function PRNGKey at 0x7f330062aac0> either PEP-noncomplia...
ERROR tests/test_serialisation.py - beartype.roar.BeartypeDecorHintNonpepException: Method equinox.nn._batch_norm.check_return() parameter "key" type hint <function PRNGKey at 0x7f330062aac0> either PEP-noncomplia...
ERROR tests/test_shared.py - beartype.roar.BeartypeDecorHintNonpepException: Method equinox.nn._batch_norm.check_return() parameter "key" type hint <function PRNGKey at 0x7f330062aac0> either PEP-noncomplia...
ERROR tests/test_stateful.py - beartype.roar.BeartypeDecorHintNonpepException: Method equinox.nn._batch_norm.check_return() parameter "key" type hint <function PRNGKey at 0x7f330062aac0> either PEP-noncomplia...
ERROR tests/test_str2jax.py - beartype.roar.BeartypeDecorHintNonpepException: Method equinox.nn._batch_norm.check_return() parameter "key" type hint <function PRNGKey at 0x7f330062aac0> either PEP-noncomplia...
ERROR tests/test_tree.py - beartype.roar.BeartypeDecorHintNonpepException: Method equinox.nn._batch_norm.check_return() parameter "key" type hint <function PRNGKey at 0x7f330062aac0> either PEP-noncomplia...
ERROR tests/test_update.py - beartype.roar.BeartypeDecorHintNonpepException: Method equinox.nn._batch_norm.check_return() parameter "key" type hint <function PRNGKey at 0x7f330062aac0> either PEP-noncomplia...
ERROR tests/test_vmap.py - beartype.roar.BeartypeDecorHintNonpepException: Method equinox.nn._batch_norm.check_return() parameter "key" type hint <function PRNGKey at 0x7f330062aac0> either PEP-noncomplia...
ERROR tests/test_while_loop.py - beartype.roar.BeartypeDecorHintNonpepException: Method equinox.nn._batch_norm.check_return() parameter "key" type hint <function PRNGKey at 0x7f330062aac0> either PEP-noncomplia...
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 32 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
================================================================================ 32 errors in 43.94s =================================================================================
```

The error message (similar for all failures):
```
E   beartype.roar.BeartypeDecorHintNonpepException: Method equinox.nn._batch_norm.check_return() parameter "key" type hint <function PRNGKey at 0x7f330062aac0> either PEP-noncompliant or currently unsupported by @beartype.
```

--- a/equinox/nn/_batch_norm.py
+++ b/equinox/nn/_batch_norm.py
@@ -4,7 +4,7 @@ from typing import Optional, Union
 import jax
 import jax.lax as lax
 import jax.numpy as jnp
-from jaxtyping import Array, Bool, Float
+from jaxtyping import Array, Bool, Float, PRNGKeyArray
 
 from .._module import field
 from ._sequential import StatefulLayer
@@ -111,7 +111,7 @@ class BatchNorm(StatefulLayer):
         x: Array,
         state: State,
         *,
-        key: Optional["jax.random.PRNGKey"] = None,  # pyright: ignore
+        key: Optional[PRNGKeyArray] = None,
         inference: Optional[bool] = None,
     ) -> tuple[Array, State]:
         """**Arguments:**

--- a/equinox/nn/_spectral_norm.py
+++ b/equinox/nn/_spectral_norm.py
@@ -111,7 +111,7 @@ class SpectralNorm(StatefulLayer, Generic[_Layer]):
         x: Array,
         state: State,
         *,
-        key: Optional["jax.random.PRNGKey"] = None,  # pyright: ignore
+        key: Optional[PRNGKeyArray] = None,
         inference: Optional[bool] = None
     ) -> tuple[Array, State]:
         """**Arguments:**

